tests/test.js

```diff
@@ -1,58 +1,55 @@
 /* eslint-env jest */
-
+const { BeginTransactionCommand, CommitTransactionCommand, ExecuteStatementCommand, RDSDataClient, RollbackTransactionCommand } = require('@aws-sdk/client-rds-data');
+const { NodeHttpHandler } = require('@smithy/node-http-handler');
+const { mockClient } = require('aws-sdk-client-mock');
+const client = require('../index');
 const constants = require('./constants');
+require('aws-sdk-client-mock-jest');
+
+const RDSDataService = mockClient(RDSDataClient);
+
+async function init (config) {
+  // We need to call the client in order to grab hold of the constructor values
+  const knex = require('knex')(config);
 
-const RDSDataService = require('aws-sdk/clients/rdsdataservice');
+  RDSDataService.on(ExecuteStatementCommand).resolves(constants.ALL_QUERY_RESPONSE_DATA);
 
-const {
-  mockExecuteStatement,
-  mockExecuteStatementPromise,
-  mockBeginTransaction,
-  mockBeginTransactionPromise,
-  mockCommitTransaction,
-  mockCommitTransactionPromise,
-  mockRollbackTransaction,
-  mockRollbackTransactionPromise
-} = RDSDataService;
+  await knex.select('*').from('foo');
 
-jest.mock('aws-sdk/clients/rdsdataservice');
+  knex.destroy();
+}
 
 beforeEach(() => {
-  RDSDataService.mockClear();
-  mockExecuteStatement.mockClear();
-  mockExecuteStatementPromise.mockReset();
-  mockBeginTransaction.mockClear();
-  mockBeginTransactionPromise.mockReset();
-  mockCommitTransaction.mockClear();
-  mockCommitTransactionPromise.mockReset();
-  mockRollbackTransaction.mockClear();
-  mockRollbackTransactionPromise.mockReset();
+  RDSDataService.reset();
 });
 
 describe('SDK configuration tests', () => {
-  test('Sets TCP Keep Alive when no sdkConfig set', () => {
-    require('knex')({
-      client: require('..'),
+  test('Sets TCP Keep Alive when no sdkConfig set', async () => {
+    await init({
+      client,
       connection: {
         database: constants.DATABASE,
         resourceArn: constants.AURORA_CLUSTER_ARN,
         secretArn: constants.SECRET_ARN
       }
     });
 
-    expect(RDSDataService).toHaveBeenCalledTimes(1);
```

Comment:
It’s important to ensure the number of times methods, like the RDSDataService constructor, have been called. I like the simplicity of many of the testing changes you’ve made, but we need to find a way to keep validating this aspect.

For what it’s worth, these checks were added in my testing when I realized that I misunderstood certain knex code paths and found some of my functions were getting called multiple times when they should have been called only once. These checks are important :).